kubernetes/samples/scripts/terraform/vcn.tf

```diff
@@ -0,0 +1,414 @@
+  }
+
+  # optional search domain.
+  # options {
+  #   type = "SearchDomain"
+  #   search_domain_names = [ "abc.com" ]
+  # }
+}
+
+/*
+ *Configures the default security list.
+ *
+ */
+resource "oci_core_default_security_list" "oke-default-security-list" {
+  manage_default_resource_id = "${oci_core_virtual_network.oke-vcn.default_security_list_id}"
+  display_name               = "${var.cluster_name}-default-security-list"
+
+  // allow outbound tcp traffic on all ports
+  egress_security_rules {
+    destination = "0.0.0.0/0"
+    protocol    = "all"
+  }
+
+  // allow inbound ssh traffic
+  ingress_security_rules {
+    protocol  = "6"         // tcp
+    source    = "0.0.0.0/0"
+    stateless = false
+
+    tcp_options {
+      "min" = 22
+      "max" = 22
+    }
+  }
+
+  // allow inbound icmp traffic of a specific type
+  ingress_security_rules {
+    protocol = 1
+    source   = "0.0.0.0/0"
+
+    icmp_options {
+      "type" = 3
+      "code" = 4
+    }
+  }
+}
+
+/*
+ * Security list for the worker subnets.
+ *  - Stateless ingress/egress rule-pairs for the worker subnets. this lets traffic between the worker
+ *    nodes flow freely. Stateless rule.
+ *  - Contains a stateful rule to allow traffic to the internet - like for pulling docker images from 
+ *    DockerHub
+ *  - Conatins two ingress rules to allow SSH traffic from OCI Cluster service.
+ */
+resource "oci_core_security_list" "oke-worker-security-list" {
+  compartment_id = "${var.compartment_ocid}"
+  display_name   = "${var.cluster_name}-Workers-SecList"
+  vcn_id         = "${oci_core_virtual_network.oke-vcn.id}"
+
+  egress_security_rules = [
+    {
+      destination = "0.0.0.0/0"
+      protocol    = "6"         // outbound TCP to the internet
+      stateless   = false
+    },
+    {
+      destination = "${var.vcn_cidr_prefix}.10.0/24"
+      protocol    = "all"
+      stateless   = true
+    },
+    {
+      destination = "${var.vcn_cidr_prefix}.11.0/24"
+      protocol    = "all"
+      stateless   = true
+    },
+    {
+      destination = "${var.vcn_cidr_prefix}.12.0/24"
+      protocol    = "all"
+      stateless   = true
+    },
+  ]
+
+  ingress_security_rules = [
+    {
+      # Intra VCN traffic - this lets the 3 subnets in teh 3 ADs tak to each other without restriction.
+      # These are stateless, so they need to be accompanied by stateless egress rules.
+      stateless = true
+
+      protocol = "all"
+      source   = "${var.vcn_cidr_prefix}.10.0/24"
+    },
+    {
+      stateless = true
+      protocol  = "all"
+      source    = "${var.vcn_cidr_prefix}.11.0/24"
+    },
+    {
+      stateless = true
+      protocol  = "all"
+      source    = "${var.vcn_cidr_prefix}.12.0/24"
+    },
+    {
+      # ICMP 
+      protocol = 1
+      source   = "0.0.0.0/0"
+
+      icmp_options {
+        "type" = 3
+        "code" = 4
+      }
+    },
+    {
+      # OCI Cluster service
+      protocol  = "6"             // tcp
+      source    = "130.35.0.0/16"
```

Comment:
what's this for?

Reply:
I believe (based on the comment in the script snippet) that this one is to allow the Clusters service to ssh into nodes to set them up during nodepool creation.  These ingress rules either came with the OKE/TF examples, or are (or used to be) required as part of the seclist setup.   However, I believe this is changing as OKE moves to support private subnets.

Generally speaking, these scripts should not be considered for creating production clusters, without more of a review.  They were mainly used to create dev/test clusters.